### DISCUSSION TITLE
Infrastructure ideas

### DISCUSSION THREAD

**Discussion by eddiejaoude** [EddieHubCommunity/LinkFree#3737] [Ideas]
Currently LinkFree is running on Digital Ocean app and managed MongoDB, but only Eddie has access.

Logs have been moved to Logtail where the maintainers can get access.

What ideas would help improve the infrastructure and maintenance?

One idea is to move back to Kubernetes on Civo using Pulumi to managed all the infrastructure, this was started in the `infrastructure` repo but not completed.

**Comment by loftwah** [comment]
I have a quick draft ready: How's this?

Currently, LinkFree is running on DigitalOcean and is managed by MongoDB, but only one person, Eddie, has access. After an outage this morning, improving the infrastructure and maintenance is essential to minimize the risk of future outages.

One idea is to move back to Kubernetes on Civo using Pulumi to manage all the infrastructure. This was started in the infrastructure repo but has yet to be completed. This approach would provide more control and flexibility over the infrastructure. Additionally, implementing an option at the DNS level, such as having a secondary Kubernetes cluster running the main branch of LinkFree, would allow for accessible domain switching in case of an outage. This would minimize downtime and allow the team to focus on resolving the issue without worrying about uptime.

Other ideas to improve the infrastructure and maintenance include:

1.  Having multiple people with access to the DigitalOcean account and MongoDB to ensure that someone is always available to address any issues.
2. Implement a monitoring and alerting system to detect and respond to any issues quickly.
3.  Setting up a disaster recovery plan to quickly recover from a significant outage.
4.  Regularly review and update the infrastructure and software to ensure they are up-to-date and secure.
5.  Having regular tests/drills to ensure that the team is prepared to handle real-world outages.
6.  Having a documented incident response plan that outlines the steps to be taken in case of an outage.
7.  Conducting a retrospective after an outage is a good idea to improve the infrastructure and maintenance. A retrospective is a meeting where the team reflects on the events during the outage, what went well, what didn't go well, and what can be improved. This can help identify areas where the team needs to improve, such as communication, incident response, and infrastructure management.

Overall, taking proactive steps such as the above can help minimize the risk of future outages and ensure that the team is prepared to handle them when they occur.

**Comment by loftwah** [comment]
Please feel free to modify and/or change any of that btw

**Comment by eddiejaoude** [comment]
As we have moved to Vercel and plan to be there for the foreseeable future, I will close this discussion
